### PR TITLE
Make settings.py optional

### PR DESCRIPTION
**Pull request by miguel-ortiz-marin**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description
- Hacktoberfest PR 👻  !
- Currently settings.py must be provided for a Kedro Project. kedro new always ship a settings.py with commented default. We should update the functionality so that when a user deleted settings.py, kedro run still works.
- Closes https://github.com/kedro-org/kedro/issues/3050

## Development notes
- Changed `test_settings.py` and `framework/project/__init__.py`
- The core Idea is to check if the file exists or not with `importlib.util.find_spec` before trying to validate it .
- I ran previous `test_settings` and included a new test and mock `mock_package_name_without_settings_file` and `test_validate_settings_without_settings_file`.
- I installed a local version of the repo with `pip install -e .` and then ran a `kedro new --starter=spaceflights`. Then I validated that `kedro run` worked with, and without a `settings.py` file.
- mypy doesn't find importlib.util so I added a # type: ignore in respective lines

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by ankatiyar** [comment]
@miguel-ortiz-marin Thanks for working on this, I know you haven't opened it up for review yet but make sure your commits are all signed off! Follow the instructions here - https://github.com/kedro-org/kedro/pull/3163/checks?check_run_id=17608910410

**Comment by miguel-ortiz-marin** [comment]
Hi @ankatiyar nice catch! I had written git commit -S instead of -s haha. 

I now pushed new commits that are passing the unit tests but the CI is failing nonetheless after updating the branch from main ⁉️ 

I'm going to mark it ready for review! I'll stay tuned for the comments

Thank you! 👻

**Comment by ankatiyar** [comment]
Did a quick manual test and things seem to be working fine and all the tests are passing 🙌🏾 but the code coverage is not complete. Could you look into this @miguel-ortiz-marin? https://github.com/kedro-org/kedro/actions/runs/6486331734/job/17633747121?pr=3163

**Comment by miguel-ortiz-marin** [comment]
@ankatiyar Awesome! I was missing one line in a unit test. 

It looks now like the test pass in linux but build is failing in windows 🤔

File: kedro/framework/project/__init__.py [hunk 1]
```diff
@@ -272,8 +272,12 @@ def validate_settings():
...
             "Kedro command line interface."
         )
-
-    importlib.import_module(f"{PACKAGE_NAME}.settings")
+    # Check if file exists, if it does, validate it.
+    if importlib.util.find_spec(f"{PACKAGE_NAME}.settings") is not None:  # type: ignore
+        importlib.import_module(f"{PACKAGE_NAME}.settings")
+    else:
+        logger = logging.getLogger(__name__)
+        logger.warning("No settings.py found, defaults will be used.")
```
Comment (merelcht) [inline kedro/framework/project/__init__.py:280]: Minor suggestion to get nicer logs highlighting: 
```suggestion
        logger.warning("No 'settings.py' found, defaults will be used.")
```

**Review by merelcht — APPROVED** [review]
Thanks for this contribution @miguel-ortiz-marin ! I left one minor comment and also wanted to ask if you can add this change to the release notes. It should go under the next non-breaking release `0.18.14`. 

You can ignore the breaking tests on CircleCI. As long as all the tests on Github Actions run everything is fine 😄

**Review by AhdraMeraliQB — APPROVED** [review]
Thanks for picking this up! ⭐

**Comment by miguel-ortiz-marin** [comment]
Looks like all's set! Thank you! 🚀

**Comment by astrojuanlu** [comment]
Congrats @miguel-ortiz-marin ! 🎉

**Comment by miguel-ortiz-marin** [comment]
Thankss! This was my first open source contribution !! To many more 🏎️ !

**Pull request merged**